### ISSUE TITLE
Option -nl and dependencies

### ISSUE THREAD

**Issue by francoisthire** [Deducteam/Dedukti#103]
File "A.dk" 
```
A : Type.

def f : A -> A -> A.

[x] f x x --> x.
```
It can be compiled with 
```
dkcheck -nl -e A.dk
```

Then in "B.dk"
```
def B := A.A.

0 : A.A.

#EVAL (A.f 0 0).
```
It evaluates the last line to `0`. In other words, the guarantee of not using the option "-nl" is that only the current file does not declare/use non-linear rules. Is this behavior wanted?

**Comment by francoisthire** [comment]
Good question. I think we should remove it too. Any opinion about this
@rafoo?

Le mer. 22 août 2018 à 09:00, Frédéric Blanqui <notifications@github.com> a
écrit :

**Comment by rafoo** [comment]
I have nothing against making `-nl` the default behaviour but for performance (matching a linear rule is more efficient) and meta-theoretic (proving confluence is easier) reasons I think we need a linearity checker.

**Comment by rlepigre** [comment]
It is of course better to avoid non-left-linear rules, but that should be the responsibility of the user to use them or not. In particular, there is no way for the implementation to know whether non-left-linearity is essential or not for a given rewriting rule. And anyway, it is in general not very natural to write non-left-linear reduction rules (we are not used to that because this is not allowed in standard programming languages). So when one writes one, that can hardly be a mistake.

**Comment by francoisthire** [comment]
Fixed in #107 .
